services-custom/s3-transfer-manager/README.md

```diff
@@ -21,88 +22,131 @@ First, you need to include the dependency in your project.
 Note that you need to replace `${awsjavasdk.version}` with the latest
 SDK version.
 
-### Instantiate the transfer manager
+### Instantiate the S3 Transfer Manager
 
 You can instantiate the transfer manager easily using the default settings:
 
 ```java
-S3TransferManager tm = S3TransferManager.create();
+S3TransferManager transferManager = S3TransferManager.create();
 ```
 
-If you wish to configure settings, we recommend using the builder instead:
+If you wish to configure settings, or use an underlying CRT-based S3 client you have already constructed, 
+we recommend using the builder instead:
 
 
 ```java
-S3TransferManager tm =
+S3AsyncClient s3AsyncClient =
+    S3AsyncClient.crtBuilder()
+                 .credentialsProvider(DefaultCredentialsProvider.create())
+                 .region(Region.US_WEST_2)
+                 .targetThroughputInGbps(20.0)
+                 .minimumPartSizeInBytes(8 * MB)
+                 .build();
+
+S3TransferManager transferManager =
     S3TransferManager.builder()
-                     .s3AsyncClient(S3CrtAsyncClient.builder()
-                                                    .credentialsProvider(credentialProvider)
-                                                    .region(Region.US_WEST_2)
-                                                    .targetThroughputInGbps(20.0)
-                                                    .minimumPartSizeInBytes(8 * MB))
+                     .s3Client(s3AsyncClient)
                      .build();
 ```
 
-### Upload a file to S3
+### Transfer a single object
 
-To upload a file to S3, you just need to provide the source file path and the `PutObjectRequest` that should be used for the upload:
+#### Upload a file to S3 and log the upload’s process with a TransferListener
+To upload a file to Amazon S3, you need to provide the source file path and a PutObjectRequest specifying the target bucket and key.
 
 ```java
-FileUpload upload =
-    tm.uploadFile(u -> u.source(Paths.get("myFile.txt"))
-                        .putObjectRequest(p -> p.bucket("bucket").key("key")));
-upload.completionFuture().join();
-```
+S3TransferManager transferManager = S3TransferManager.create();
 
-### Download an S3 object to a file
+UploadFileRequest uploadFileRequest =
+    UploadFileRequest.builder()
+                     .putObjectRequest(req -> req.bucket("bucket").key("key"))
+                     .addTransferListener(LoggingTransferListener.create())
```

Comment:
We're adding a listener here, but not demonstrating how "log the upload’s process with a TransferListener" works.

Reply:
Added a comment there. LMK if there's a better way

Reply:
I tried to add a section for TransferListeners, but I feel like I was just copying pasting code from single-object upload/download sections, so I ended up just adding additional notes for TransferListener. LMK what you think